### PR TITLE
Git for Beginners

### LINKED ISSUE #220: Write reference page for `git`
https://c4cs.github.io/commands/git

### PR DESCRIPTION
**Pull request by smacphers**
# Advanced HW 7 submission

Closes #220 

Name: Steven Macpherson
Uniqname: smacpher

### CODE REVIEW

File: _commands/tools/git.md [hunk 1]
```diff
@@ -25,6 +25,286 @@ git
 
 ----------------------
 
+Git For Beginners
+=================
+This section of the Git page is for those of us that are actually in EECS 280/281, meaning you have little to no experience with Git and are more worried about figuring out the 
+difference between staging and committing things than the neat little things 
+you can do to make Git as personalized as possible.
+
+What is Git?
```
Comment (mterwill) [inline _commands/tools/git.md]: [Markdown](https://daringfireball.net/projects/markdown/) will also support varying sizes of headers using `#`, `##`, `###`, etc.

**Comment by ethanmad** [comment]
Wow! Nice work!

Can I, as another student, suggest linking to "[Atomic Commits](https://www.freshconsulting.com/atomic-commits/) (or similar) and "[How to Write a Git Commit Message](http://chris.beams.io/posts/git-commit/)" in your commit section?

It's really nice to work with other people when they understand why to write atomic commits and how to describe their changes.

**Comment by smacphers** [comment]
Yeah, for sure.

**Comment by mterwill** [comment]
Thanks Steven! Merging this in + giving you credit.

@ethanmad – good article. Would be happy to accept another PR if you'd like to add it in!

**Comment by ethanmad** [comment]
@mterwill Will do!

File: _commands/tools/git.md [hunk 2]
```diff
@@ -141,7 +143,110 @@ q, which is something that took me an embarassingly long time to figure out. Now
...
+Creating New Features in A Sandbox (Sort Of)
+--------------------------------------------
+
+One of the many advantages of git is that you can work on a new feature or implementation 
+without messing up your current working code. Say I decided I want to re-write one of 
+my functions so that it is tail-recursive instead of just recursive. I would create a new 
+branch with the name of tail at my current commit by typing:
+
+~~~ bash
+git branch -d tail
```
Comment (mterwill) [inline _commands/tools/git.md]: Good catch!

**Comment by smacphers** [comment]
Yeah thanks for pointing that out. Sorry about that!

**Pull request merged**